sdk/src/azure/core/az_mqtt5_rpc.c

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/az_mqtt5_rpc.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/core/internal/az_log_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <stdio.h>
+
+#if defined(TRANSPORT_MOSQUITTO)
+#include <mosquitto.h>
+#endif
+
+#include <azure/core/_az_cfg.h>
+
+AZ_NODISCARD bool az_span_topic_matches_sub(az_span sub, az_span topic)
+{
+  bool ret;
+#if defined(TRANSPORT_MOSQUITTO)
+  if (MOSQ_ERR_SUCCESS
+      != mosquitto_topic_matches_sub((char*)az_span_ptr(sub), (char*)az_span_ptr(topic), &ret))
+  {
+    ret = false;
+  }
+#else // TRANSPORT_MOSQUITTO
+  (void)sub;
+  (void)topic;
+  ret = false;
+#endif
+  return ret;
+}
+
+AZ_NODISCARD bool az_mqtt5_rpc_status_failed(az_mqtt5_rpc_status status)
+{
+  return (status < 200 || status >= 300);
+}
+
+AZ_NODISCARD az_result az_rpc_get_topic_from_format(
+    az_span model_id,
+    az_span executor_client_id,
+    az_span invoker_client_id,
+    az_span command_name,
+    az_span format,
+    az_span out_topic,
+    int32_t* out_topic_length)
+{
+  const az_span model_id_key = AZ_SPAN_FROM_STR("{serviceId}");
```

Comment:
Is there any situation where any of these keys would appear more than once in the topic format? Or is it documented that that is not supported?

Reply:
ooh, interesting question. I'll have to find out if that's supported or not

Reply:
It's not explicitly disallowed right now - I added a comment for now calling out the limitation in our code, but we might have to support this scenario in the future